src/Microsoft.IdentityModel.Tokens/Validation/TokenTypeValidationResult.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Contains the result of validating the TokenType of a <see cref="SecurityToken"/>.
+    /// The <see cref="TokenValidationResult"/> contains a collection of <see cref="ValidationResult"/> for each step in the token validation.
+    /// </summary>
+    internal class TokenTypeValidationResult : ValidationResult
+    {
+        private Exception? _exception;
+
+        /// <summary>
+        /// Creates an instance of <see cref="TokenTypeValidationResult"/>.
+        /// </summary>
+        /// <paramref name="type"/> is the type against which the token was validated.
+        public TokenTypeValidationResult(string? type)
+            : base(ValidationFailureType.ValidationSucceeded)
+        {
+            Type = type;
+            IsValid = true;
```

Comment:
If 'ValidationResult' is 'ValidationFailureType.ValidationSucceeded' then is it  valid in all conditions? If yes, then this set logic should be in `ValidationResult` constructor.

If not, then a constructor of `ValidationResult` should be updated to include the `IsValid` parameter. 

Two step constructions should be avoided wherever possible.

Reply:
The `ValidationResult` is still a work in progress so, we will make additional changes on a separate PR also, since we are returning IsValid as true when no validation has been performed this won't affect the current state of `ValidationResult` and the Type parameter still needs to be assigned even if we remove this constructor, so I do not see how we would avoid the two-step construction of the object. However, it could be simplified to only declaring type and setting the IsValid on `ValidationResult` base. Thanks for the feedback @SaurabhMSFT, I'll definitely keep it mind for when we have a finalized `ValidationResult`.